--- who/coreutils-8.24-lava-safe/src/who.c.orig	2019-01-21 15:49:54.182055534 +0900
+++ who/coreutils-8.24-lava-safe/src/who.c	2017-09-30 23:21:18.000000000 +0900
@@ -1,3 +1,6 @@
+#include <config.h>
+#include <getopt.h>
+#include <stdio.h>
 extern unsigned int lava_get(unsigned int) ;
 void lava_set(unsigned int bn, unsigned int val);
 static unsigned int lava_val[1000000];
@@ -5,17 +8,14 @@
 void lava_set(unsigned int bug_num, unsigned int val) { lava_val[bug_num] = val; }
 unsigned int lava_get(unsigned int bug_num);
 unsigned int lava_get(unsigned int bug_num) {
-#if 0
 #define SWAP_UINT32(x) (((x) >> 24) | (((x) & 0x00FF0000) >> 8) | (((x) & 0x0000FF00) << 8) | ((x) << 24))
     if (0x6c617661 - bug_num == lava_val[bug_num] ||
         SWAP_UINT32(0x6c617661 - bug_num) == lava_val[bug_num]) {
-        printf("Successfully triggered bug %d, crashing now!\n", bug_num);
-        exit(0);
+        dprintf(1, "Successfully triggered bug %d, crashing now!\n", bug_num);
     }
     else {
         //printf("Not successful for bug %d; val = %08x not %08x or %08x\n", bug_num, lava_val[bug_num], 0x6c617661 + bug_num, 0x6176616c + bug_num);
     }
-#endif
     return lava_val[bug_num];
 }
 /* GNU's who.
@@ -43,10 +43,6 @@
    idle: -u
 */
 
-#include <config.h>
-#include <getopt.h>
-#include <stdio.h>
-
 #include <sys/types.h>
 #include "system.h"
 
